lib/NodeType.js

```diff
@@ -26,9 +26,10 @@ const NodeType = {
   FILE_PATH: 'FILE_PATH',
   PARENTHESIS: 'PARENTHESIS',
   TYPE_QUERY: 'TYPE_QUERY',
+  KEY_QUERY: 'KEY_QUERY',
   IMPORT: 'IMPORT',
 };
 
 /** @typedef {keyof typeof NodeType} Type */
 
-module.exports = NodeType;
+module.exports = /** @type {{readonly [T in Type]: T}} */ (NodeType);
```

Comment:
that type seems circular (`NodeType -> Type -> NodeType`). Do you start getting weird implicit-any or circularity errors if you turn on `"strict": true` in tsconfig?

Reply:
This is not circular, as this doesn’t affect the definition of the `NodeType` object, only the `NodeType.js` file export.

Reply:
Oh, right, never mind. I was thinking NodeType was an *alias* for `module.exports`, but it's not.

I've spent too much of my life trying to check code like this:

```js
module.exports = NodeType = {
  ...
}
NodeType.x = 1
NodeType.y = 2
// etc
```